### PR TITLE
Sync with upstream release v25.0

### PR DESCRIPTION
**Pull request by russell-islam**
Sync with upstream release v25.0. This PR is still not complete. I can boot with PCI transport. virtio-mmio seems not working yet.
Also, I haven't tested this on aarch64.

changes:
1. Able to build using rust 1.59.0 (nre version)
2. Fixes tons of conflicts and new changes upstream
3. new PCI structure for activations needs more investigations
4. Need some investigations on mmio transport
5. many more

### CODE REVIEW

**Comment by russell-islam** [comment]
I was able to boot using mmio transport on x86_64. Need more time to fix all the fmt and clippy issues.

**Comment by russell-islam** [comment]
Since we are targetting for rust 1.59 only I have removed all the nightly and beta tests. Now the PR is ready to be merged. @liuw @NunoDasNeves

**Review by liuw — APPROVED** [review]

File: arch/src/aarch64/fdt.rs [hunk 1]
```diff
@@ -99,6 +99,7 @@ pub fn create_fdt<T: DeviceInfoForFdt + Clone + Debug, S: ::std::hash::BuildHash
     gic_device: &Arc<Mutex<dyn Vgic>>,
     initrd: &Option<InitramfsConfig>,
     pci_space_info: &[PciSpaceInfo],
+    #[cfg(feature = "pci_support")]
     numa_nodes: &NumaNodes,
```
Comment (NunoDasNeves) [inline arch/src/aarch64/fdt.rs]: are NUMA nodes tied to the PCI feature somehow?

File: arch/src/aarch64/mod.rs [hunk 2]
```diff
@@ -146,6 +146,7 @@ pub fn configure_system<T: DeviceInfoForFdt + Clone + Debug, S: ::std::hash::Bui
     pci_space_info: &[PciSpaceInfo],
     virtio_iommu_bdf: Option<u32>,
     gic_device: &Arc<Mutex<dyn Vgic>>,
+    #[cfg(feature = "acpi")]
```
Comment (NunoDasNeves) [inline arch/src/aarch64/mod.rs]: the other references to NUMA nodes are gated by the PCI feature, not ACPI

File: arch/src/aarch64/fdt.rs (hunk 1 above)
Comment (russell-islam) [reply arch/src/aarch64/fdt.rs]: No. It is ACPI feature gated. Let me take a look again.

**Review by NunoDasNeves — APPROVED** [review]

**Pull request merged**